diff --git a/lua/plenary/async/tests.lua b/lua/plenary/async/tests.lua
index 4895773..d9a8346 100644
--- a/lua/plenary/async/tests.lua
+++ b/lua/plenary/async/tests.lua
@@ -7,7 +7,7 @@ M.describe = function(s, async_func)
 end
 
 M.it = function(s, async_func)
-  it(s, util.will_block(async_func))
+  it(s, util.will_block(async_func, 60000)) -- TODO: use test timeout
 end
 
 M.before_each = function(async_func)
diff --git a/lua/plenary/async/util.lua b/lua/plenary/async/util.lua
index 3e6f033..d224f7d 100644
--- a/lua/plenary/async/util.lua
+++ b/lua/plenary/async/util.lua
@@ -40,9 +40,12 @@ M.block_on = function(async_function, timeout)
   return unpack(ret)
 end
 
-M.will_block = function(async_func)
+---@see M.block_on
+---@param async_function Future
+---@param timeout number
+M.will_block = function(async_function, timeout)
   return function()
-    M.block_on(async_func)
+    M.block_on(async_function, timeout)
   end
 end
 
